src/Query.jl

```diff
@@ -216,9 +216,22 @@ function select_callback(
 end
 
 "Execute a query, streaming the resulting blocks through a channel."
+function select_df_channel(
+    query::AbstractString,
+    sock::ClickHouseSock = connect();
+    csize = 0,
+    kwargs...,
+)::Channel{DataFrame}
+    Channel(ctype = DataFrame, csize = csize) do ch
+        select_callback(sock, query; kwargs...) do row
+            put!(ch, DataFrame!(row))
+        end
+    end
+end
+
 function select_channel(
-    sock::ClickHouseSock,
-    query::AbstractString;
+    query::AbstractString,
+    sock::ClickHouseSock = connect();
```

Comment:
While this might have been the better design, this is interface breaking and will require code changes by all users of the lib. Such interface breaking changes should be discussed properly and shouldn't be introduced in a PR that has a completely different general goal.

Same goes for all the other select variants below.